src/common/services/session/sessionModal.spec.js

```diff
@@ -0,0 +1,60 @@
+import angular from 'angular';
+import 'angular-mocks';
+import module from './sessionModal.service';
+
+describe( 'sessionModalService', function () {
+  beforeEach( angular.mock.module( module.name ) );
+  let sessionModalService, $uibModal;
+
+  beforeEach( inject( function ( _sessionModalService_, _$uibModal_ ) {
+    sessionModalService = _sessionModalService_;
+    $uibModal = _$uibModal_;
+    // Spy On $uibModal.open and return mock object
+    spyOn( $uibModal, 'open' ).and.returnValue( {result: angular.noop} );
+  } ) );
+
+  it( 'should be defined', () => {
+    expect( sessionModalService ).toBeDefined();
+  } );
+
+  describe( 'open', () => {
+    it( 'should be defined', () => {
+      expect( sessionModalService.open ).toBeDefined();
+    } );
+
+    it( 'should open \'sign-in\' by default', () => {
+      sessionModalService.open();
+      expect( $uibModal.open ).toHaveBeenCalled();
+      expect( $uibModal.open.calls.count() ).toEqual( 1 );
+      expect( $uibModal.open.calls.argsFor( 0 )[0].resolve.state() ).toEqual( 'sign-in' );
+    } );
+
+    it( 'should allow options', () => {
+      sessionModalService.open( 'sign-up', {backdrop: false, keyboard: false} );
+      expect( $uibModal.open ).toHaveBeenCalled();
+      expect( $uibModal.open.calls.count() ).toEqual( 1 );
+      let options = $uibModal.open.calls.argsFor( 0 )[0];
+      expect( options.resolve.state() ).toEqual( 'sign-up' );
+      expect( options.backdrop ).toEqual( false );
+      expect( options.keyboard ).toEqual( false );
```

Comment:
You could do something like this but since there's that `state()` function call, idk that you can  incorporate it in the `toHaveBeenCalledWith` so it might be cleaner to leave it as is. I'll share the code anyways...
```
expect($uibModal.open).toHaveBeenCalledWith(jasmine.objectContaining({
    backdrop: false,
    keyboard: false
}));